File: Two Pointers/ContainerWithMostWater.py
def maxArea(self, height: List[int]) -> int:
    
    l = 0 
    r = len(height) - 1
    
    w = 0
    h = 0
    area = 0
    
    maxArea = 0
    
    while l < r:
        
        w = r - l
        h = min(height[l], height[r])
        area = h * w
        
        maxArea = max(area, maxArea)
        
        if (height[l] < height[r]):
            l += 1
            
        else:
            r -= 1
            
    return maxArea

File: 02_Two_Pointers/ContainerWithMostWater.java
public class ContainerWithMostWater {
    
    public int maxArea(int[] heights){

        int l = 0, r = heights.length - 1;  // left and right pointers

        int area = 0, maxArea = 0;

        int h, w;   // height and width, respectively

        while (l < r){

            // Calculate height, width, and area
            h = Math.min(heights[l], heights[r]);
            w = r = l;
            area = h * w;

            // Check maxArea
            if (area > maxArea)
                maxArea = area;

            // Shift pointer
            if (heights[l] > heights[r])
                r--;

            else
                l++;

        }

        return maxArea;

    }

}
